pages/post/[slug].js

```diff
@@ -157,23 +176,33 @@ export default function Post() {
                     <div className="my-3 text-[0.875rem]">
                       {post.readingTime} mins
                     </div>
+
                     <div className="mt-4 text-[1rem] md:text-[1.010rem] list-none">
-                      {indexContent.map((content, index) => (
-                        <div
-                          key={index}
-                          onClick={handleClick}
-                          className="my-3 hover:underline"
-                          dangerouslySetInnerHTML={{
-                            __html: content,
-                          }}
-                        />
-                      ))}
+                      {indexContent.map((content, index) => {
+                        var href = content
+                          .match(/href="#(.*?)"/g)[0]
+                          .slice(7, -1);
+                        return (
+                          <div
+                            key={index}
+                            onClick={handleClick}
+                            className={`my-3 hover:bg-gradient-to-r from-pink-300 to-orange-300 hover:text-transparent hover:bg-clip-text ${
+                              activeId === href
+                                ? "relative bg-gradient-to-r bg-clip-text text-transparent after:absolute after:bottom-0 after:left-0 after:w-full after:h-[1px] after:bg-gradient-to-r from-pink-300 to-orange-300"
```

Comment:
I think underline's width should be equal to text length only instead of in full div